### DISCUSSION TITLE
HI I need help to change the CC1101 channel

### DISCUSSION THREAD

**Discussion by Minga312** [jgromes/RadioLib#511] [Q&A]
HI ,
I don't find in the source code where I can change my channel,
in others examples I see that it's possible but I need some help to make it here

**Comment by jgromes** [comment]
I would suggest trying the `setFrequency` method:

https://github.com/jgromes/RadioLib/blob/7f09fc430aceff61a85112a7f2c7264e25bb43de/src/modules/CC1101/CC1101.h#L680-L687

**Reply by Minga312** [reply]
thank you so much , I din't realize that I could do it manually, 
How is the range that you get with this lib with cc1101? 
I change the settings so PA would be 10 dbm but in a open place I only was able to get 100 m of distance, do you have some other orientation that I can make it the range larger ?

**Reply by jgromes** [reply]
Simply put, the range you will get is inversly proportional to the data rate - higher data rate = lower range. You haven't mentioned what speed you are using, but lowering it could help.

The type of antenna on both ends of the link will also make a huge difference.

**Reply by Minga312** [reply]
![1651060063917](https://user-images.githubusercontent.com/70277063/165510149-f3f5a161-c87e-4ca2-9b30-6066ec10a102.jpg)
 Hi I'm using this antena, 32kps and 434 Mhz, probably it's my antena do you have others to recommend me ?

**Reply by jgromes** [reply]
That is probably not tuned very well, a quarter/half wave dipole will do a better job. There are loads of tutorials online online on making them, just out of plain old wire.

32 kbps is also quite fast for sub-GHz transceivers like CC1101, so unless you really need that speed, I would suggest slowing down, which is going to improve your receiver sensitivity (very roughly: half speed = twice the sensitivity).
